timemachine/cpp/src/context.cu

```diff
@@ -63,8 +63,11 @@ Context::multiple_steps(const std::vector<double> &lambda_schedule, int store_du
     if (store_x_interval <= 0) {
         throw std::runtime_error("store_x_interval <= 0");
     }
-    int du_dl_buffer_size = (lambda_schedule.size() + store_du_dl_interval - 1) / store_du_dl_interval;
-    int x_buffer_size = (lambda_schedule.size() + store_x_interval - 1) / store_x_interval;
+    if (lambda_schedule.size() % store_x_interval != 0) {
+        std::cout << "warning:: length of lambda_schedule modulo store_x_interval does not equal zero" << std::endl;
```

Comment:
I'm having trouble thinking of cases where this shouldn't just be an error. It seems like this would result in extra effort that's unobservable from the calling code?

Reply:
>I'm having trouble thinking of cases where this shouldn't just be an error. It seems like this would result in extra effort that's unobservable from the calling code?

Oh -- not always an error, still observable from calling code!

Consider an idiom [like](https://github.com/proteneer/timemachine/blob/ba7114f6147b55d6ff972ffc901dc32eb39dda9d/timemachine/md/moves.py#L110-L111)

```python
x_before = ...
_ = ctxt.multiple_steps(x_before, ...)
x_after = ctxt.get_x_t()
```

(where you discard the result of multiple_steps, and access the updated state of the context). The value of `store_x_interval` will be irrelevant to the behavior of this code.

The change in this PR should mostly obviate that idiom though, and make it more natural to use the return value of `multiple_steps`, rather than accessing the updated state of `ctxt`

Reply:
Ah, makes sense. Thanks for the clarification!